tests/cli/bootsys/test_mgmt_power.py

```diff
@@ -177,3 +183,160 @@ def test_mgmt_ncns_skip_prompt_power_off(self):
         do_power_off_ncns(self.args)
         self.mock_prompt_continue.assert_not_called()
         self.mock_do_mgmt_shutdown_power.assert_called_once()
+
+
+class TestDoMgmtShutdownPower(unittest.TestCase):
+    """Tests for the do_mgmt_shutdown_power function."""
+
+    def setUp(self):
+        """Set up mocks."""
+        self.username = 'admin'
+        self.password = 'password'
+
+        self.mock_args = mock.Mock()
+        self.ncn_shutdown_timeout = 1200
+        self.ipmi_timeout = 60
+
+        self.mock_get_and_verify_ncn_groups = mock.patch(
+            'sat.cli.bootsys.mgmt_power.get_and_verify_ncn_groups').start()
+        self.mock_get_ssh_client = mock.patch(
```

Comment:
Should also probably mock the following:

```suggestion
        self.mock_filtered_host_keys = mock.patch(
            'sat.cli.bootsys.mgmt_power.FilteredHostKeys').start()
        self.mock_get_ssh_client = mock.patch(
```